cns/fsnotify/fsnotify.go

```diff
@@ -5,41 +5,45 @@ import (
 	"io"
 	"io/fs"
 	"os"
+	"runtime"
 	"sync"
 	"time"
 
 	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/cns/hnsclient"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
 	"github.com/fsnotify/fsnotify"
 	"github.com/pkg/errors"
-	"go.uber.org/zap"
 	"golang.org/x/sync/errgroup"
 )
 
-type releaseIPsClient interface {
+type endpointManager interface {
 	ReleaseIPs(ctx context.Context, ipconfig cns.IPConfigsRequest) error
+	GetEndpoint(ctx context.Context, endpointID string) (*restserver.GetEndpointResponse, error)
 }
 
 type watcher struct {
-	cli  releaseIPsClient
-	path string
-	log  *zap.Logger
-
+	cli           endpointManager
+	path          string
 	pendingDelete map[string]struct{}
 	lock          sync.Mutex
+	cnsconfig     *configuration.CNSConfig
```

Comment:
don't pass the cns config in here, this package does not need to know all of that 🙁

Reply:
Correct. I'll only pass the IntiFromCNI and ManagedEndpointState.